### PR TITLE
[TASK-6748] [TASK-7307] [TASK-7306] fix: qa session issues

### PR DESCRIPTION
**Pull request by jjramirezn**
This PR contains several bugfixes and improvements

- Correct wallet connect and peanut sign in handling
- Show existing passkeys when signing a transacion
- Fix max value on token selector for peanut wallet
- Fix text on already pay request
- Add copy button to wallet cards in home

## Summary by CodeRabbit

- **New Features**
	- Introduced a new method for selecting external wallets, streamlining wallet connection processes.
	- Enhanced wallet balance display with improved formatting for better clarity.
	- Added functionality to copy wallet addresses to the clipboard for user convenience.
	- Updated user interface text for clarity in the payment status component.
	- Improved error handling and user feedback mechanisms during wallet connection and login processes.
	- Enhanced recipient address handling based on wallet type for improved user experience.
	- Integrated new hooks for better state management and notifications.

- **Bug Fixes**
	- Enhanced error handling across various components, providing more specific user feedback.

- **Documentation**
	- Updated component logic descriptions to reflect new functionalities and user interactions.

- **Style**
	- Minor adjustments to UI elements for enhanced user experience.

### CODE REVIEW

File: src/app/(mobile-ui)/layout.tsx [hunk 1]
```diff
@@ -186,7 +186,7 @@ const Layout = ({ children }: { children: React.ReactNode }) => {
                         loading={isLoggingIn}
                         disabled={isLoggingIn}
                         onClick={() => {
-                            handleLogin()
+                            handleLogin().then(signInModal.close)
```
[the remark this answers is not in the document]
Comment (jjramirezn) [reply src/app/(mobile-ui)/layout.tsx]: done

File: src/app/(mobile-ui)/layout.tsx [hunk 2]
```diff
@@ -205,11 +205,7 @@
                         variant="dark"
                         shadowType="secondary"
                         onClick={() => {
-                            web3Modal.open().finally(() => {
-                                if (isConnected) {
-                                    signInModal.close()
-                                }
-                            })
+                            web3Modal.open().then(selectExternalWallet).finally(signInModal.close)
```
[the remark this answers is not in the document]
Comment (jjramirezn) [reply src/app/(mobile-ui)/layout.tsx]: done

File: src/utils/balance.utils.ts [hunk 3]
```diff
@@ -101,3 +102,7 @@ export function calculateValuePerChain(balances: IUserBalance[]): ChainValue[] {
     }
     return result
 }
+
+export function printableUsdc(baseUnitsAmount: bigint): string {
+    return Number(formatUnits(baseUnitsAmount, 6)).toFixed(2)
+}
```
Comment (Hugo0) [inline src/utils/balance.utils.ts:108]: Thought: seems a bit hardcoded

File: src/components/Request/Pay/Views/GeneralViews/AlreadyPaid.view.tsx [hunk 4]
```diff
@@ -21,7 +21,7 @@ export const AlreadyPaidLinkView = ({ requestLinkData }: { requestLinkData: _con
     return (
         <Card className="shadow-none sm:shadow-primary-4">
             <Card.Header className="text-center">
-                <Card.Title>Sorry, this link has already been paid.</Card.Title>
+                <Card.Title>This link previously contained:</Card.Title>
```
Comment (Hugo0) [inline src/components/Request/Pay/Views/GeneralViews/AlreadyPaid.view.tsx]: In `main` this line is like this:

https://github.com/peanutprotocol/peanut-ui/blob/6c6cc0484e07ca3fd562c90fa6815b07b7ef61cd/src/components/Request/Pay/Views/GeneralViews/AlreadyPaid.view.tsx#L29

`            <label className="text-h2">Payment Receipt</label>`

We should keep that and not overwrite!

File: src/components/Home/WalletCard.tsx [hunk 5]
```diff
@@ -75,13 +74,20 @@ export function WalletCard({ type, onClick, ...props }: WalletCardProps) {
...
                         </div>
+                        <Icon
+                            name="content-copy"
+                            className="absolute bottom-0 right-0 h-5 w-5 hover:opacity-80"
+                            fill="white"
+                            onClick={(e: MouseEvent<SVGElement>) => {
+                                e.stopPropagation()
+                                navigator.clipboard.writeText(wallet.address)
+                            }}
+                        />
```
Comment (Hugo0) [inline src/components/Home/WalletCard.tsx]: In my opinion this is bad from a UX perspective - you click the copy thing, but nothing happens, visually. It's not clear that anything was copied.

We should make it clear for the user that smth happened. I'll make a discord thread

**Review by Hugo0 — COMMENTED** [review]
see comment below

**Comment by Hugo0** [comment]
Behavioural QA on https://peanut-6590depp0-squirrellabs.vercel.app/request/create:
1. It says connect wallet even though I am signed in and have a peanut smart wallet

image

2. idg the point of this connected thing: 
image

3. new wallets don't get connected / clicking on byow doesn't work:
image
3.1 same thing when I try to fulfill a payment request with an external wallet, after clicking connect nth happens
image

File: src/components/Request/Pay/Views/GeneralViews/AlreadyPaid.view.tsx (hunk 4 above)
Comment (jjramirezn) [reply src/components/Request/Pay/Views/GeneralViews/AlreadyPaid.view.tsx]: done

File: src/components/Home/WalletCard.tsx (hunk 5 above)
Comment (jjramirezn) [reply src/components/Home/WalletCard.tsx]: added toast info

File: src/utils/balance.utils.ts (hunk 3 above)
Comment (jjramirezn) [reply src/utils/balance.utils.ts:108]: I think is ok for now, PW is only usdc and usdc is 6 decimals

**Comment by jjramirezn** [comment]
Added some requested fixes
- Show error when adding already existing external account
-  Hide and autocomplete recipient input on receive
- Directly login in send flows if peanut wallet is selected but not connected

**Comment by jjramirezn** [comment]
Added error when trying to claim not supported token

File: src/app/(mobile-ui)/layout.tsx [hunk 6]
```diff
@@ -205,11 +212,14 @@ const Layout = ({ children }: { children: React.ReactNode }) => {
...
-                                }
-                            })
+                            web3Modal
+                                .open()
+                                .then(selectExternalWallet)
+                                .catch((e) => {
+                                    console.error(e)
+                                    toast.error('Error connecting wallet')
+                                })
+                                .finally(signInModal.close)
```
Comment (Hugo0) [inline src/app/(mobile-ui)/layout.tsx:222]: nice!

File: src/components/Home/HomeHeader.tsx [hunk 7]
```diff
@@ -27,19 +28,21 @@ const HomeHeader = () => {
                     <p>www.peanut.me/</p>
                     <p className="text-h4">{username}</p>
```
Comment (Hugo0) [inline src/components/Home/HomeHeader.tsx:29]: Not sure i understand the logic here

**Pull request merged**